### DISCUSSION TITLE
Get product options in storefront

### DISCUSSION THREAD

**Discussion by SebastianBotez** [lunarphp/lunar#1543] [Q&A]
Hello.
I have an issue where I don't understand how can I get the product options in store front.
Any ideea how the logic should be?

For now our store has an example product with variants that has 2 options (size and color).

<img width="377" alt="Screenshot 2024-02-08 at 20 47 04" src="https://github.com/lunarphp/lunar/assets/10911720/6530c792-5b42-441a-aef7-6f976878e9f0">

In the store front page, how can we get the available options for the specified product ? I'm thinking that the options should be dynamic. Maybe one product has size and color and other has only size. If I query from lunar_product_options table I get all the options available even if one particular product has all of the options or no.
Thank you

**Comment by alecritson** [comment]
Which version are you currently using? I'm assuming 0.7/0.8?

**Reply by SebastianBotez** [reply]
Hello @alecritson. I'm using "lunarphp/lunar": "^0.7.5"

**Reply by alecritson** [reply]
Ok cool, so yeah in `0.7` it's not that straightforward and something that's being addressed in `1.0`. In the meantime you could do something like this:

```php
$product = \Lunar\Models\Product::find(1);

$productOptions = \Lunar\Models\ProductOption::with([
    'values' => function ($query) use ($product) {
        $query->whereHas('variants', function ($relation) use ($product) {
            $relation->where('product_id', $product->id);
        });
    },
])->orderBy('position')->get()->reject(function ($option) {
    return $option->values->isEmpty();
})->values();
```

So what we're doing is:

1. Fetching all product options
2. Eager loading the values in, based on whether they are associated with a variant that belongs to the product
3. Order by the product option position
4. Filter out any Product Options which don't have any values

Again, this is changing in 1.x to be a bit more streamlined, but hopefully this gets you going for now.

**Reply by SebastianBotez** [reply]
Oh yes thank you. Now I understand the logic. So basically we are doing the query as we need it. I will wait for now to upgrade to 1.x because i saw that it's not stable. Thank you very much for the help!
